x/gov/keeper/proposal.go

```diff
@@ -229,6 +229,18 @@ func (keeper Keeper) DeleteProposal(ctx context.Context, proposalID uint64) erro
 		if err != nil {
 			return err
 		}
+
+		err = keeper.QuorumCheckQueue.Walk(ctx, nil, func(key collections.Pair[time.Time, uint64], _ v1.QuorumCheckQueueEntry) (bool, error) {
+			// we need to traverse the entire queue as we do not know with certainty the value of
+			// the first part of the key (the time part)
+			if key.K2() == proposalID {
+				return false, keeper.QuorumCheckQueue.Remove(ctx, key)
```

Comment:
```suggestion
				return true, keeper.QuorumCheckQueue.Remove(ctx, key)
```

So the walk stops once we found the key.